backend/src/test/java/de/wirvsvirus/trackyourbed/HospitalServiceTest.java

```diff
@@ -0,0 +1,318 @@
+import de.wirvsvirus.trackyourbed.entity.HospitalCapacity;
+import de.wirvsvirus.trackyourbed.excpetion.resource.NoSuchHospitalException;
+import de.wirvsvirus.trackyourbed.persistence.HospitalRepository;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@DisplayName("Tests for HospitalService")
+@ExtendWith(MockitoExtension.class)
+class HospitalServiceTest {
+
+  @Mock
+  private CreateNewHospitalMapper createNewHospitalMapper;
+
+  @Mock
+  private HospitalRepository hospitalRepository;
+
+  @Mock
+  private HospitalDtoMapper hospitalDtoMapper;
+
+  @Mock
+  private CapacityService capacityService;
+
+  @Mock
+  private HospitalCapacityDtoMapper hospitalCapacityDtoMapper;
+
+  private HospitalService hospitalService;
+
+  @BeforeEach
+  private void setUp() {
+    this.hospitalService = new HospitalService(
+        hospitalRepository,
+        createNewHospitalMapper,
+        hospitalDtoMapper,
+        capacityService,
+        hospitalCapacityDtoMapper);
+  }
+
+  @Test
+  @DisplayName("Should call dependencies and return expected result when called.")
+  void shouldCreateNewHospital() {
+    // GIVEN
+    final String name = "hospital";
+    final int maxCapacity = 23;
+    final String lon = "11.11";
+    final String lan = "12.11";
+    final CreateNewHospital request = new CreateNewHospital()
+        .setMaxCapacity(maxCapacity)
+        .setLon(lon)
+        .setLat(lan)
+        .setName(name);
+    final Hospital hospital = new Hospital();
+    final HospitalDto expected = new HospitalDto();
+
+    when(createNewHospitalMapper.dtoToEntity(request)).thenReturn(hospital);
+    when(hospitalRepository.save(any(Hospital.class))).thenReturn(hospital);
+    when(hospitalDtoMapper.entityToDto(any(Hospital.class))).thenReturn(expected);
+
+    // WHEN
+    final HospitalDto actual = hospitalService.createHospital(request);
+
+    // THEN
+    assertSame(expected, actual);
+    verify(createNewHospitalMapper).dtoToEntity(argThat(r -> {
+      assertSame(request, r);
+      return true;
+    }));
+    verify(hospitalRepository).save(argThat(b -> {
+      assertSame(hospital, b);
+      return true;
+    }));
+    verify(hospitalDtoMapper).entityToDto(argThat(b -> {
+      assertSame(hospital, b);
+      return true;
+    }));
+  }
+
+  @Test
+  void shouldDeleteHospital() {
+    //Given
+    doNothing().when(hospitalRepository).deleteById(any(UUID.class));
+
+    //When
+    hospitalService.deleteHospital(UUID.randomUUID());
+
+    //then
+    verify(hospitalRepository, times(1)).deleteById(any(UUID.class));
```

Comment:
- The `UUID` parameter should be verified
- an explicit `times(1)` is not necessary